Core/Character.cs

```diff
@@ -1,9 +1,128 @@
 ﻿using System;
+using System.Collections.Generic;
 
 namespace PCSharpGen.Core
 {
     public class Character
     {
+        public Character(BaseGameRules rules)
+        {
+            Rules = rules;
+        }
+
+        public BaseGameRules Rules { get; }
+
         public string Name { get; set; }
+
+        public Dictionary<string, CharacterVariable> Variables { get; } = new Dictionary<string, CharacterVariable>(StringComparer.InvariantCultureIgnoreCase);
+
+        public CharacterVariable GetVariable(string name)
+        {
+            if (Variables.TryGetValue(name, out var result))
+                return result;
+            result = new CharacterVariable(this, new List<GameValue>());
+            Variables.Add(name, result);
+            return result;
+        }
+
+        public BonusBuilder SetVariableBase(string name, int value)
+        {
+            var fixedValue = new FixedValue("BASE", value);
+            if (Variables.TryGetValue(name, out CharacterVariable stat))
+            {
+                stat.Replace("BASE", fixedValue);
+            }
+            else
+            {
+                Variables.Add(name, new CharacterVariable(this, fixedValue));
+            }
+
+            return new BonusBuilder(this, name);
+        }
+
+        public BonusBuilder AddToVariable(string name, string type, int value)
+        {
+            GetVariable(name).Add(new FixedValue(type, value));
+            return new BonusBuilder(this, name);
+        }
+
+        public BonusBuilder AddToVariable(string name, int value)
+        {
+            return AddToVariable(name, name, value);
+        }
+
+        public BonusBuilder AddReferenceToVariable(string name, string type, string referenceStatistic)
```

Comment:
This function name is a bit weird. I wasn't sure what it did in the beginning. maybe these should be replaced with something like `character.GetVariable(name).Add(type, character.GetVariable(other))` or something like that. I see the implementation is mostly that, but the method is strange being on Character, because it doesn't actually do anything to the character.

Reply:
Well, it _does_ do something to the character.  After calling this method, the characters variable "name" will have an extra modifier.

I have the "Define" method... maybe I can update that to "Modify" and just make everything use the builder, so it's a lot clearer?

Reply:
I tried this, it's a lot clearer, and maps more to what the "Formula" thing look line in the LUA.